### ISSUE TITLE
gospel pps detaches documentation comments when below and followed by a specification

### ISSUE THREAD

**Issue by n-osborne** [ocaml-gospel/gospel#266]
This behaviour can be reproduced as follow:

```
$ cat > ./bug_pps.mli << EOF
> val f : int -> int
> (** blah *)
> (*@ y = f x *)
> EOF
```
 Then we have:
```
$ ocamlc -dsource bug_pps.mli
val f : int -> int[@@ocaml.doc " blah "]
```
Which is the expected behaviour.

But:
```
$ ocamlc -pp "gospel pps" -dsource bug_pps.mli
val f : int -> int[@@gospel {| y = f x |}]
```
Where the documentation comment has been lost.

**Comment by jmid** [comment]
Good catch Nicolas! :raised_hands:

**Comment by n-osborne** [comment]
After a discussion with @shym, I've tried:  
```
$ cat > ./bob.mli << EOF
> val f : int -> int[@@ocaml.doc " blah "]
> (*@ y = f x *)
> EOF
$ ocamlc -pp "gospel pps" bob.mli -dsource
val f : int -> int[@@ocaml.doc " blah "][@@gospel {| y = f x |}]
```

I've also tried this:
```
$ cat > . bob.mli << EOF
> val f : int -> int
> (*@ y = f x *)
> (** blah *)
> EOF
$ ocamlc -pp "gospel pps" bob.mli -dsource
val f : int -> int[@@gospel {| y = f x |}][@@ocaml.doc " blah "]
```

The limitation (if we write the gospel specification below a documentation comment, the latter will be detached and ignored if we run `gospel pps` before compiling) may well find its origin in the OCaml parser itself (the way it handles documentation comments).

One way out (again, a suggestion from @shym), would be to make `gospel pps` turn documentation comments into `ocaml.doc` attributes when they occur between the OCaml declaration and the Gospel specifications. Or just print them after the gospel attribute, as suggested by the second case (so that we reuse the OCaml parser to turn documentation comments into proper attributes).
